geoSegment/geoSegment.js

```diff
@@ -233,47 +233,55 @@ async function processSessionSeriesItems(items, segments) {
       continue;
     }
 
-    // Filter virtual sessions
-    let sessionSeriesPhysicalLocationGeo;
-    if (item.data.location && item.data.location.geo) {
-      sessionSeriesPhysicalLocationGeo = item.data.location.geo;
-    } else if (item.data['beta:affiliatedLocation'] && item.data['beta:affiliatedLocation'].geo) {
-      sessionSeriesPhysicalLocationGeo = item.data['beta:affiliatedLocation'].geo;
-    } else {
-      sessionSeriesPhysicalLocationGeo = null;
-      // if (item.data.eventAttendanceMode === "https://schema.org/OnlineEventAttendanceMode") {
-      // } else {
-      //   continue;
-      // }
-    }
-
     // Filter out high-frequency session data
     if (item.data['beta:presentAsSlots'] === true || (item.data.superEvent && item.data.superEvent['beta:presentAsSlots'] === true)) {
       continue;
     }
 
+    const sessionSeriesGeoSegments = (() => {
+      let tempSegments = [];
+      let sessionSeriesPhysicalLocationGeo;
+
+      const generateSegments = (physicalLocationGeo) => {
+        let temp = [];
+        for (const segment of segments) {
+          const segmentRadiusInMeters = segment.radius * 1000;
+          const distanceBetweenGeos = geolib.getDistance(
+            { latitude: physicalLocationGeo.latitude, longitude: physicalLocationGeo.longitude },
+            { latitude: segment.latitude, longitude: segment.longitude },
+          );
+          if (distanceBetweenGeos <= segmentRadiusInMeters) {
+            temp.push(segment.identifier);
+          }
+        }
+
+        return temp;
+      }
+
+      if (!item.data.eventAttendanceMode || item.data.eventAttendanceMode === "https://schema.org/OfflineEventAttendanceMode" || item.data.eventAttendanceMode === "https://schema.org/MixedEventAttendanceMode") { // offline event
+        if (item.data.location && item.data.location.geo) {
+          sessionSeriesPhysicalLocationGeo = item.data.location.geo;
+          tempSegments = generateSegments(sessionSeriesPhysicalLocationGeo);
```

Comment:
Also the `sessionSeriesPhysicalLocationGeo` variable is I think redundant. You could instead replace these two lines with `return generateSegments(item.data.location.geo);`